File: 1790-check-if-one-string-swap-can-make-strings-equal/1790-check-if-one-string-swap-can-make-strings-equal.java
class Solution {
    public boolean areAlmostEqual(String s1, String s2) {
          char arr[] = s2.toCharArray();
        char ans[] = s2.toCharArray();
        if(s1.equals(s2))
            return true;
        for(int i=0;i<arr.length;i++){
            int j = i+1;
            while(j<s2.length()){
                char temp = arr[i];
                arr[i] = arr[j];
                arr[j] = temp;
                String str = new String(arr);
                if(str.equals(s1))
                    return true;
                else{
                    arr[i] = ans[i];
                    arr[j] = ans[j];
                }
                j++;
            }
        }
        return false;
    }
}

File: 1790-check-if-one-string-swap-can-make-strings-equal/1790-check-if-one-string-swap-can-make-strings-equal.cpp
class Solution {
public:
    bool areAlmostEqual(string s1, string s2) {
              int ct=0,a1=-1,a2,b1,b2;
        for(int k=0;k<s1.length();k++)
        {
            if(s1[k]!=s2[k])
            {
                ct++;
                if(ct>2)
                    return false;
                if(a1==-1)
                {
                    a1=s1[k]-'a';
                    b1=s2[k]-'a';
                }
                a2=s1[k]-'a';
                b2=s2[k]-'a';
            }
        }
        if(ct==0)
            return true;
        if(ct<2)
            return false;
        if(b1==a2 && b2==a1)
            return true;
        return false;  
    }
};